src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -17,6 +18,10 @@ namespace Microsoft.AspNetCore.Mvc.Razor
     /// <inheritdoc />
     public class RazorPageActivator : IRazorPageActivator
     {
+        // Delegates to create ViewDataDictionary of TModel
+        private delegate ViewDataDictionary ViewDataCreateFromSource(ViewDataDictionary source);
+        private delegate ViewDataDictionary ViewDataCreateFromModel(IModelMetadataProvider metadataProvider, ModelStateDictionary modelState);
```

Comment:
Suggestions for naming 

`ViewDataCreateFromSource` -> `CreateViewDataNested`
`ViewDataCreateFromModel` -> `CreateViewDataRoot`